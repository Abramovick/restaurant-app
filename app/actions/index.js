import * as types from '../constants/Actiontypes';

const receiveProducts = products => ({
  type: types.RECEIVE_PRODUCTS,
  products
});

const addToCartFn = product => ({
  type: types.ADD_TO_CART,
  product
});

const addToken = token => ({
  type: types.ADD_TOKEN,
  token
});

const login = loginState => ({
  type: types.LOGIN_FAIL,
  loginState
});

const checkOut = () => ({
  type: types.CHECKOUT,
});

const increaseQuantity = product => ({
  type: types.INCREASE_ITEM_QUANTITY,
  product
});

const decreaseQuantity = product => ({
  type: types.DECREASE_ITEM_QUANTITY,
  product
});

const clearAuth = () => ({
  type: types.CLEAR_AUTH,
});

export const addProducts = products => (dispatch) => {
  dispatch(receiveProducts(products));
};
export const addTokenToStore = token => (dispatch) => {
  dispatch(addToken(token));
};
export const addToCart = product => (dispatch) => {
  dispatch(addToCartFn(product));
};

export const increaseItemQuantity = product => (dispatch) => {
  product.quantity += 1;
  dispatch(increaseQuantity(product));
};

export const decreaseItemQuantity = product => (dispatch) => {
  product.quantity -= 1;
  dispatch(decreaseQuantity(product));
};

export const clearCart = () => (dispatch) => {
  dispatch(checkOut());
};

export const removeAuth = () => (dispatch) => {
  dispatch(clearAuth());
};

export const loginFail = loginState => (dispatch) => {
  dispatch(login(loginState));
};
