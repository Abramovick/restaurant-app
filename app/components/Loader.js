import React from 'react';
import PropTypes from 'prop-types';
import { View, Modal, ActivityIndicator } from 'react-native';
import { LoaderStyles as styles } from '../styles/styles';

const Loader = (props) => {
  const {
    loading,
    closeModal
  } = props;
  return (
    <Modal
      transparent
      onRequestClose={() => closeModal}
      animationType={'slide'}
      visible={loading}
    >
      <View style={styles.modalBackground}>
        <View style={styles.activityIndicatorWrapper}>
          <ActivityIndicator
            size="large"
            color="#F0E68C"
            animating={loading} />
        </View>
      </View>
    </Modal>
  );
};

Loader.propTypes = {
  products: PropTypes.array,
  onClickImage: PropTypes.func,
  loading: PropTypes.bool,
  closeModal: PropTypes.func,
};

export default Loader;
