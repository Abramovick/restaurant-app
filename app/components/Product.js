import React from 'react';
import { View, Image, Text, TouchableOpacity } from 'react-native';
import StarRating from 'react-native-star-rating';
import { Icon, Button } from 'native-base';
import Modal from 'react-native-modal';
import PropTypes from 'prop-types';
import { ShopStyles as styles, ModalStyle } from '../styles/styles';


const Product = ({
  params,
  imgWidth,
  imgHeight,
  starCount,
  isModalVisible,
  showModal,
  hideModal,
  onStarRatingPress
}) =>
<View style={styles.productView}>
<View>
  <Image
  source={{ uri: params.ImageUrl }}
  style={{ width: imgWidth, height: imgHeight }}
  />
</View>
<View>
  <Text style={styles.productName}>{ params.Name }</Text>
</View>
<View style={{ flexDirection: 'row', display: 'flex' }}>
  <View>
    <Text style={styles.productPrice}>Price: N{ params.Price }</Text>
    <View style={{ marginTop: 10 }}>
      <StarRating
        disabled={false}
        maxStars={5}
        rating={starCount}
        starSize={16}
        starColor={'#F7C04C'}
        selectedStar={rating => onStarRatingPress(rating)}
      />
    </View>
  </View>
  <View style={{ flexDirection: 'row', flex: 1, justifyContent: 'flex-end' }}>
    <TouchableOpacity
     onPress={showModal}
      style={{
        borderWidth: 1,
        borderColor: 'rgba(0,0,0,0.2)',
        alignItems: 'center',
        justifyContent: 'center',
        width: 50,
        height: 50,
        backgroundColor: '#F7C04C',
        borderRadius: 100,
      }}
    >
      <Icon name='md-cart' />
    </TouchableOpacity>
    <Modal isVisible={isModalVisible}>
      <View style={ModalStyle.modalContent}>
        <Text style={ModalStyle.heading}>Added to Cart</Text>
        <Icon name='ios-checkmark-circle-outline' style={{ fontSize: 100, color: '#F7C04C' }}/>
        <Button onPress={hideModal} color="#F7C04C" style={ModalStyle.okButton}>
          <Text style={ModalStyle.buttonText}>     OK    </Text>
        </Button>
      </View>
    </Modal>
  </View>
</View>
<View style={styles.hr} />
<Text style={styles.description}> Description </Text>
<Text style={styles.descriptionText}>{params.Description}</Text>
</View>;

Product.propTypes = {
  params: PropTypes.object,
  imgWidth: PropTypes.number,
  imgHeight: PropTypes.number,
  starCount: PropTypes.number,
  isModalVisible: PropTypes.bool,
  showModal: PropTypes.func,
  hideModal: PropTypes.func,
  onStarRatingPress: PropTypes.func

};

export default Product;
