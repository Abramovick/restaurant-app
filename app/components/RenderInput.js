import React from 'react';
import PropTypes from 'prop-types';
import { Input, Item, Icon } from 'native-base';
import { HomeScreenStyles as styles } from '../styles/styles';

const RenderInput = ({
  placeholder,
  meta: { touched, error },
  input: { onChange, ...restInput }
}) =>
    <Item style={(touched && error) ?
      [styles.inputView, styles.errorStyle] : styles.inputView}>
      {(touched && error) && <Icon style={styles.searchIcon} name='alert'/>}
      <Input
      style={styles.input}
      onChangeText={onChange}
      {...restInput}
      placeholder={placeholder}
      placeholderTextColor="white"/>
    </Item>;

RenderInput.propTypes = {
  placeholder: PropTypes.string,
};

RenderInput.propTypes = {
  meta: PropTypes.object,
  input: PropTypes.object,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
};

export default RenderInput;
