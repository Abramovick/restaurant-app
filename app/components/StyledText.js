import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

export const StyledText = props =>
  <Text
  {...props}
    style={[props.style, { fontFamily: 'SinkinSans-200XLight' }]}
  />;

StyledText.propTypes = {
  style: PropTypes.object
};
