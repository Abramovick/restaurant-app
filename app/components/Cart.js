import React from 'react';
import {
  ScrollView,
  View,
  Image
} from 'react-native';
import {
  Button,
  Icon,
  Card,
  CardItem,
} from 'native-base';
import ModalDropdown from 'react-native-modal-dropdown';
import PropTypes from 'prop-types';
import { StyledText as Text } from './StyledText';
import { CartScreenStyles as styles } from '../styles/styles';


const Cart = ({ items }) =>
  <View style={styles.container}>
    <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
      <View style={styles.main}>
        <Card >
          <CardItem style={styles.cardItem}>
            <View style={styles.cardItemView}>
              <Text style={styles.font12}>ITEM</Text>
            </View>
            <View style={[styles.cardItemView, styles.quantity]}>
              <Text style={styles.font12}>No</Text>
            </View>
            <View style={[styles.cardItemView, styles.price]}>
              <Text style={styles.font12}>Price</Text>
            </View>
            <View style={[styles.cardItemView, styles.subTotal]}>
              <Text style={styles.font12}>SubTotal</Text>
            </View>
          </CardItem>
        </Card>
        {items.map((item, index) => (
          <Card key={index}>
            <CardItem style={styles.cardItem}>
              <View style={styles.content}>
                <Image source={{ uri: item.url }} style={styles.Image} />
                <Text style={styles.cartProductName}>
                {item.name}
                </Text>
              </View>
              <View style={styles.cardContent}>
                <ModalDropdown
                defaultValue='  1  '
                style={styles.modal}
                textStyle={styles.modalText}
                dropdownTextStyle ={styles.dropdownText}
                options={['  1  ', '  2  ', '  3  ', '  4  ', '  5  ']}/>
              </View>
              <View style={styles.cardContent}>
                <Text style={styles.cartBody}>Price</Text>
              </View>
              <View style={styles.subTotal}>
                <Text style={styles.cartBody}>SubTotal</Text>
              </View>
            </CardItem>
          </Card>
          ))}
        <Button color="#F7C04C" style={styles.button} block iconLeft>
            <Icon name='md-checkmark'/>
            <Text style={styles.buttonText}>    CHECKOUT   </Text>
        </Button>
      </View>
    </ScrollView>
  </View>;

Cart.propTypes = {
  items: PropTypes.array
};

export default Cart;
