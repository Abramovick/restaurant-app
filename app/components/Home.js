import React from 'react';
import {
  ScrollView,
  TouchableOpacity,
  View,
} from 'react-native';
import {
  Button,
  Icon,
} from 'native-base';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import { HomeScreenStyles as styles } from '../styles/styles';
import RenderInput from '../components/RenderInput';
import { StyledText as Text } from './StyledText';
import Loader from './Loader';

const Home = ({
  handleSubmit,
  loginUser,
  handleSignupPress,
  loginFail,
  loading,
  visibleHeight,
  handleFacebookLogin,
  handleGoogleLogin
}) =>
<View style={{ height: visibleHeight, backgroundColor: '#D57E56' }}>
<ScrollView keyboardShouldPersistTaps={'handled'}>
    <View style={styles.main}>
      <View style={styles.logoContainer}>
        <Text style={styles.largeText}>Sign  In</Text>
        <Text style={styles.logoText}>Restaurant</Text>
      </View>
      <Field
        name="email"
        component={RenderInput}
        placeholder="E-mail"
      />
      <Field
        name="password"
        component={RenderInput}
        placeholder="Password"
        secureTextEntry
      />
      {loginFail &&
      <View style={styles.loginFailView}>
        <Icon name='alert' style={styles.loginFail}/>
        <Text style={styles.loginFailText}>Invalid Username or Password</Text>
      </View>}
      <Button onPress={handleSubmit(loginUser)} color="#F7C04C" style={styles.button} block iconLeft>
        <Icon name='md-checkmark'/>
        <Text style={styles.buttonText}>    Login</Text>
      </Button>
      <Text style={styles.forgotPassword}>
        Forgot Password?
      </Text>
      <View>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.forgotPassword}>Do not have account?  </Text>
          <TouchableOpacity onPress={handleSignupPress}>
            <Text style={styles.signUpText}>Sign Up here</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
    <View style={styles.helpContainer}>
      <View style={{ flexDirection: 'row' }}>
        <View style={styles.lineStyle}/>
        <Text style={styles.forgotPassword}>
          Or Log In with
        </Text>
        <View style={styles.lineStyle}/>
      </View>
    </View>
    <View style={styles.buttonsView} >
      <Button
      onPress={handleFacebookLogin}
      style={styles.facebookLoginBtn}
      color="#F7C04C"
      block
      iconLeft>
          <Icon name='logo-facebook' />
          <Text style={styles.buttonText}>  Facebook</Text>
      </Button>
      <Button
      onPress={handleGoogleLogin}
      color="#F7C04C"
      style={styles.googleLoginBtn}
      block
      iconLeft>
          <Icon name='logo-google' />
          <Text style={styles.buttonText}>  Google</Text>
      </Button>
    </View>
    <View style={styles.paddingView}/>
    { loading && <Loader loading={loading} />
}
    </ScrollView>
</View>;

Home.propTypes = {
  handleSubmit: PropTypes.func,
  loginUser: PropTypes.func,
  handleSignupPress: PropTypes.func,
  handleFacebookLogin: PropTypes.func,
  handleGoogleLogin: PropTypes.func,
  loginFail: PropTypes.bool,
  loading: PropTypes.bool,
  visibleHeight: PropTypes.number
};

export default Home;
