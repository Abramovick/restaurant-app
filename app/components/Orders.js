import React from 'react';
import {
  ScrollView,
  View,
  Image,
  TextInput
} from 'react-native';
import {
  Button,
  Card,
  Icon,
  CardItem,
  Content
} from 'native-base';
import lodash from 'lodash';
import moment from 'moment';
import PropTypes from 'prop-types';
import Modal from 'react-native-modal';
import StarRating from 'react-native-star-rating';
import { StyledText as Text } from './StyledText';
import { OrdersScreenStyles as styles } from '../styles/styles';

const Orders = ({
  allOrders,
  ratingStarCount,
  onReviewStarRatingPress,
  addReview,
  selectedOrder,
  openModal,
  closeModal,
  onChangeText,
  isModalVisible
}) =>
<View style={styles.view}>
  <ScrollView>
    { allOrders && allOrders.map((order, index) => (
      <Card key={index} style= {styles.card}>
        <CardItem>
          <Button
            color="#F7C04C"
            style={styles.cancelButton}
            block
            iconLeft
          >
            <Text>DATE ORDERED: {moment(order.createdAt).format('Do MMMM YYYY')} </Text>
          </Button>
        </CardItem>
        <CardItem style={styles.cardItem}>
          <Text style={styles.textStyle}> NO </Text>
          <Text style={styles.textStyle}> ITEM </Text>
          <Text style={styles.textStyle}>QUANTITY</Text>
        </CardItem>
          {
            lodash.flattenDeep(order.items).map((item, number) => (
              <CardItem key={number} style={styles.cardItem}>
                <Text style={styles.textStyle}> {number + 1}. </Text>
                <Text style={styles.textStyle}> {item.name} </Text>
                <Text style={styles.textStyle}>{item.quantity}</Text>
              </CardItem>
            ))
          }
        <CardItem style={styles.cardItem}>
          <View style={styles.viewStyle}/>
        </CardItem>
        <CardItem style={styles.cardItem}>
          <Text>TOTAL AMOUNT PAID:  {order.total}</Text>
        </CardItem>
        <CardItem style={styles.cardItem}>
          <View style={styles.viewStyle}/>
        </CardItem>
        <CardItem>
          <Button
            color='white'
            block
            iconLeft
            onPress={() => openModal(order.items)}
            style={styles.reviewButton}
          >
            <Text color='white'>ADD REVIEW</Text>
          </Button>
        </CardItem>
      </Card>
    ))}
    <Modal isVisible={isModalVisible}>
      <Content>
        <Card style={styles.card}>
          <CardItem>
            <Button
              color="#F7C04C"
              style={styles.cancelButton}
              block
              iconLeft
            >
              <Text>ADD YOUR REVIEWS </Text>
            </Button>
          </CardItem>
          { selectedOrder.map((item, number) => (
            <CardItem key={number} style={styles.reviewCardItem}>
              <Image
                source={{ uri: item.imageUrl }}
                style={styles.Image}
              />
              <View style={{ marginLeft: 10, width: '66%' }}>
                <Text style={styles.reviewText}> {item.name} </Text>
                <View style={{ width: '40%', marginTop: 10 }}>
                  <StarRating
                    disabled={false}
                    maxStars={5}
                    rating={ratingStarCount[item.id]}
                    starSize={16}
                    starColor={'#F7C04C'}
                    selectedStar={rating => onReviewStarRatingPress(rating, item.id)}
                  />
                </View>
                <TextInput
                  underlineColorAndroid='rgba(0,0,0,0)'
                  multiline
                  placeholder={'Add your reviews'}
                  style={styles.textInput}
                  onChangeText={text => onChangeText(text, item.id)}
                />
              </View>
            </CardItem>
            ))
          }
          <CardItem>
            <Button
              color='white'
              block
              iconLeft
              onPress={addReview}
              style={styles.reviewButton}
            >
              <Text color='white'>SUBMIT REVIEW</Text>
            </Button>
          </CardItem>
          <CardItem>
            <Button
              color='white'
              block
              iconLeft
              onPress={closeModal}
              style={styles.reviewButton}
            >
              <Text color='white'>CANCEL</Text>
            </Button>
          </CardItem>
        </Card>
      </Content>
    </Modal>
    {allOrders.length === 0 &&
    <Content>
      <Card style={styles.card}>
        <CardItem style={styles.cardItem}>
          <Text fontSize={15} > SORRY, YOU HAVE NO ORDERS</Text>
        </CardItem>
        <CardItem style={styles.cardItem}>
          <Icon name="ios-sad-outline" style={styles.Icon}/>
        </CardItem>
      </Card>
    </Content>
    }
  </ScrollView>
</View>;

Orders.propTypes = {
  allOrders: PropTypes.array,
  ratingStarCount: PropTypes.object,
  onReviewStarRatingPress: PropTypes.func,
  addReview: PropTypes.func,
  openModal: PropTypes.func,
  closeModal: PropTypes.func,
  isModalVisible: PropTypes.bool,
  onChangeText: PropTypes.func,
  selectedOrder: PropTypes.array
};

export default Orders;
