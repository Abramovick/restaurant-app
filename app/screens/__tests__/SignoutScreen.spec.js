import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { shallow, } from 'enzyme';
import thunk from 'redux-thunk';
import { StyledText as Text } from '../../components/StyledText';
import ConnectedSignout, { SignoutScreen } from '../SignoutScreen';
import { removeAuth } from '../../actions';

// Test objects


const navigation = {
  navigate: jest.fn(),
};

// Test suites
describe('SIGNOUTSCREEN --- Snapshot', () => {
  it('renders the sign out screen', async () => {
    const tree = renderer.create(<SignoutScreen
    removeAuth={removeAuth}
    navigation={navigation}
    />);
    expect(tree.getInstance().handleCancel()).toMatchSnapshot();
    expect(tree.getInstance().handleLogout()).toMatchSnapshot();
    expect(SignoutScreen.navigationOptions.drawerLabel())
      .toEqual(<Text
        style={{ fontFamily: 'SinkinSans-200XLight', padding: 15 }}
        >
        LogOut
        </Text>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('SIGNOUTSCREEN --- Shallow rendering + passing the store directly', () => {
  const initialState = {
    user: {
      token: 'fesfefs',
      userId: 32342342342,
      loginState: true
    }
  };
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  let store, container, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedSignout store={store}/>);
  });

  it('renders the connected component', () => {
    expect(container.length).toEqual(1);
  });

  it('renders the connected component', () => {
    wrapper = shallow(<SignoutScreen
    navigation={navigation}
    removeAuth={removeAuth}
    />);
    expect(wrapper.length).toEqual(1);
  });

  it('check action on dispatching ', () => {
    let action = '';
    store.dispatch(removeAuth());
    action = store.getActions();
    expect(action[0].type).toBe('CLEAR_AUTH');
  });
});
