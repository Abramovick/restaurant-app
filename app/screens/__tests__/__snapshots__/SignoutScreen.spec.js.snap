// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGNOUTSCREEN --- Snapshot renders the sign out screen 1`] = `undefined`;

exports[`SIGNOUTSCREEN --- Snapshot renders the sign out screen 2`] = `undefined`;

exports[`SIGNOUTSCREEN --- Snapshot renders the sign out screen 3`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "flex": 0,
              "padding": 25,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "SinkinSans-200XLight",
                "fontSize": 14,
                "marginBottom": 20,
              }
            }
          >
            Are you sure you want to logout?
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#F7C04C",
              "borderBottomWidth": null,
              "borderColor": "#F7C04C",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": 0.5,
              "elevation": 2,
              "flexDirection": "row",
              "height": 35,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontFamily": "SinkinSans-200XLight",
                "fontSize": 14,
              }
            }
          >
                LOG OUT
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "white",
              "borderBottomWidth": null,
              "borderColor": "#F7C04C",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": 0.5,
              "elevation": 2,
              "flexDirection": "row",
              "height": 35,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F7C04C",
                "fontFamily": "SinkinSans-200XLight",
                "fontSize": 14,
              }
            }
          >
                CANCEL
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
